lib/logos.js

```diff
@@ -88,25 +88,38 @@ function getSimpleIconStyle({ icon, style }) {
   return 'default'
 }
 
-function getSimpleIcon({ name, color, style }) {
+function getSimpleIcon({ name, color, style, size = 'contain' }) {
   const key = name === 'travis' ? 'travis-ci' : name.replace(/ /g, '-')
 
   if (!(key in simpleIcons)) {
     return undefined
   }
 
+  let iconSvg
+
   const svgColor = toSvgColor(color)
   if (svgColor) {
-    return svg2base64(
-      simpleIcons[key].svg.replace('<svg', `<svg fill="${svgColor}"`),
-    )
+    iconSvg = simpleIcons[key].svg.replace('<svg', `<svg fill="${svgColor}"`)
   } else {
     const iconStyle = getSimpleIconStyle({ icon: simpleIcons[key], style })
-    return simpleIcons[key].base64[iconStyle]
+    iconSvg = simpleIcons[key].styles[iconStyle]
+  }
+
+  if (size === 'auto') {
+    const { width: iconWidth, height: iconHeight } = getIconSize(key)
+
+    if (iconWidth > iconHeight) {
+      const path = resetIconPosition(simpleIcons[key].path)
+      iconSvg = iconSvg
+        .replace('viewBox="0 0 24 24"', `viewBox="0 0 24 ${iconHeight}"`)
+        .replace(/<path d=".*"\/>/, `<path d="${path}"/>`)
+    }
   }
+
+  return svg2base64(iconSvg)
 }
 
-function prepareNamedLogo({ name, color, style }) {
+function prepareNamedLogo({ name, color, style, size }) {
```

Comment:
Now that we're pretty settled on the implementation, could we add a test case for the `auto` style to https://github.com/badges/shields/blob/2a417804d54626938d6fbcbeb9659315a2f98ce2/lib/logos.spec.js#L33